spec/outputs/example_spec.rb

```diff
@@ -1 +1,20 @@
 require "logstash/devutils/rspec/spec_helper"
+require "logstash/outputs/example"
+require "logstash/codecs/plain"
+require "logstash/event"
+
+describe LogStash::Outputs::Example do
+  let(:sample_event) { LogStash::Event.new }
+
+  context "#receive" do
+    subject do
+      output = LogStash::Outputs::Example.new
+      output.register
```

Comment:
Me as always :panda_face: , I would make this something like

```
subject(:output) { LogStash::Outputs::Example.new }
before :all do
  output.register
end
```

even if your options works and it good, the solution I propose here is more semantically correct in order to know and learn where to put what and when.